File: frontend/src/api/review.ts
import { get, post } from '../utils/request';
import type { PageData } from './product';

export interface ReviewVO {
  id: number;
  user_id: number;
  username: string;
  order_id: number;
  order_item_id: number;
  product_id: number;
  product_name: string;
  rating: number;
  content: string;
  images: string[];
  reply: string;
  replied_at: string;
  created_at: string;
}

export const createReview = (data: { order_item_id: number; rating: number; content: string; images?: string[] }) =>
  post<{ message: string; review: ReviewVO }>('/reviews', data);

export const listReviews = (params?: { page?: number; page_size?: number; product_id?: number; order_id?: number; user_id?: number }) =>
  get<PageData<ReviewVO>>('/reviews', params as Record<string, unknown>);

export const replyReview = (id: number, reply: string) => post<{ message: string; review: ReviewVO }>(`/admin/reviews/${id}/reply`, { reply });

File: backend/internal/model/review.go
package model

import "time"

// Review 评价实体：订单完成后用户评价，商家可回复。
type Review struct {
	ID          uint       `gorm:"primaryKey" json:"id"`
	UserID      uint       `gorm:"index;not null" json:"user_id"`
	OrderID     uint       `gorm:"index;not null" json:"order_id"`
	OrderItemID uint       `gorm:"index;not null" json:"order_item_id"`
	ProductID   uint       `gorm:"index;not null" json:"product_id"`
	Rating      int        `gorm:"not null" json:"rating"`
	Content     string     `gorm:"type:text" json:"content"`
	Images      string     `gorm:"type:text" json:"images"`
	Reply       string     `gorm:"type:text" json:"reply"`
	RepliedAt   *time.Time `json:"replied_at"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`

	User    *User    `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Product *Product `gorm:"foreignKey:ProductID" json:"product,omitempty"`
}
